cmake/EthCompilerSettings.cmake

```
@@ -170,6 +170,10 @@ if (("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" MA
 			set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -s ALLOW_TABLE_GROWTH=1")
 			# Disable warnings about not being pure asm.js due to memory growth.
 			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-almost-asm")
+			# Increase stack size from 5MB to 16MB to prevent stack overflow issues.
+			set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -s TOTAL_STACK=16mb")
```

Apparently this setting was renamed in [3.1.25](https://github.com/emscripten-core/emscripten/blob/main/ChangeLog.md#3125---110822). Let's use the current name. I actually had a bit of a problem locating its docs before I realized this.

```suggestion
			set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -s STACK_SIZE=16mb")
```

---

When I checked a month ago, you were using Emscripten 3.1.19 to build soljson, where the new settings were not recognized. Did you update the Emscripten version?

---

Ah, right. It seemed quite old so I assumed we're already on a newer one, but apparently not: https://github.com/ethereum/solidity/blob/3dc2cf864267d4e97257d16c7626cd3e14a1e7ec/scripts/docker/buildpack-deps/Dockerfile.emscripten#L35

I guess we should update :)